litestar/serialization/msgspec_hooks.py

```diff
@@ -112,7 +115,7 @@ def default_deserializer(
             if predicate(target_type):
                 return decoder(target_type, value)
 
-    if issubclass(target_type, (Path, PurePath, ImmutableState, UUID)):
+    if issubclass(target_type, (Path, PurePath, ImmutableState, UUID, SecretString, SecretBytes)):
```

Comment:
Should raise `TypeError` for these if `value` isn't either `str` or `bytes` respectively